rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/520_fetch_fields.yml

```diff
@@ -130,16 +130,71 @@ fetch _seq_no via stored_fields:
 fetch _seq_no via fields:
 
   - do:
-      catch: "request"
+      catch: bad_request
       search:
         index: test
         body:
           fields: [ _seq_no ]
 
-  # This should be `unauthorized` (401) or `forbidden` (403) or at least `bad request` (400)
-  # while instead it is mapped to an `internal_server_error (500)`
-  - match: { status: 500 }
-  - match: { error.root_cause.0.type: unsupported_operation_exception }
+  - match: { status: 400 }
+  - match: { error.root_cause.0.type: illegal_argument_exception }
+  - match: { error.root_cause.0.reason: "error fetching [_seq_no]: Cannot fetch values for internal field [_seq_no]." }
+
+---
+fetch _source via fields:
```

Comment:
I wonder if at least for `_source` we might want to document somewhere how fetching `_source` works via fields or stored fields. Also considering that if we use synthetic source and we request it via stored fields we don't get anything back...which might be "scary" for some users...

Reply:
Thank you, @salvatore-campagna, for your feedback. I agree; some documentation would be helpful. Is there a dedicated team for the documentation part, or should I provide an update myself?

Reply:
In a case like ths I would just write it and involve someone from the the docs team to just review it.

Reply:
I am ok for not adding the test as part of this PR...as this deals only with the fields api.

Reply:
Focusing on documentation, I tested the `metadata fields` behavior when included under `fields` in search. In the following list (I am wondering if I have included all metadata fields).

```
// Return value
_id / _index / _version

// Ignored
_type / _score / _primary_term / _routing / _shard / _doc / _ignored / _meta / _ttl / _timestamp / _cluster_state / _cat / _stats / _aliases

// 400 error    
_source / _seq_no / _feature / _nested_path / _field_names
```

In the first category, the field is returned, e.g.
```
"hits": [
            {
                "_index": "test,"
                "_id": "1",
                "_score": 0.6931471,
                "fields": {
                    "_id": [
                        "1"
                    ],
                    "tags": [
                        "staging"
                    ]
                }
            }
        ]
```
In the second one, the fields are `ignored`, and in the third, a `400` error is thrown, for example,
```
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_argument_exception",
                "reason": "error fetching [_source]: Cannot fetch values for internal field [_source]."
            }
        ],
        "type": "search_phase_execution_exception",
        "reason": "all shards failed",
        "phase": "query",
        "grouped": true,
        "failed_shards": [
            {
                "shard": 0,
                "index": "test",
                "node": "1MC-EHn3TD-3XLPyUcE43A",
                "reason": {
                    "type": "illegal_argument_exception",
                    "reason": "error fetching [_source]: Cannot fetch values for internal field [_source].",
                    "caused_by": {
                        "type": "illegal_argument_exception",
                        "reason": "Cannot fetch values for internal field [_source]."
                    }
                }
            }
        ],
        "caused_by": {
            "type": "illegal_argument_exception",
            "reason": "error fetching [_source]: Cannot fetch values for internal field [_source].",
            "caused_by": {
                "type": "illegal_argument_exception",
                "reason": "error fetching [_source]: Cannot fetch values for internal field [_source].",
                "caused_by": {
                    "type": "illegal_argument_exception",
                    "reason": "Cannot fetch values for internal field [_source]."
                }
            }
        }
    },
    "status": 400
}
```
Do we need to add this behavior to the public documentation, and if so, to what extent? In my point of view, the error itself is self-explanatory, and the user can easily identify it.  What do you think? :)